source/adapters/hip/enqueue.cpp

```diff
@@ -1582,49 +1585,43 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueDeviceGlobalVariableWrite(
     if (offset + count > DeviceGlobalSize)
       return UR_RESULT_ERROR_INVALID_VALUE;
 
-    return urEnqueueUSMMemcpy(
-        hQueue, blockingWrite,
-        reinterpret_cast<void *>(reinterpret_cast<uint8_t *>(DeviceGlobal) +
-                                 offset),
-        pSrc, count, numEventsInWaitList, phEventWaitList, phEvent);
+    void *pSrc, *pDst;
+    if (CopyType == GlobalVariableCopy::Write) {
+      pSrc = ptr;
+      pDst = reinterpret_cast<void *>(
```

Comment:
Do you need `void *` cast, will it not be deduced?

Reply:
Good catch have changed